[DOCS] Update patching options and daal4py install guide (#880)
intel/scikit-learn-intelex: doc/sources/patching/patching-options.rst

    from sklearnex import patch_sklearn

>     patch_sklearn()

Reviewer: Could you please also add note here that it's required to reimport in this case? 

And also same not to main readme.md

.. important::

>     Scikit learn import must be done **after** these lines, otherwise the patching will not affect the original scikit-learn estimators.

Reviewer 2: ```suggestion
    You have to import scikit-learn **after** these lines. Otherwise, the patching will not affect the original scikit-learn estimators.
```

- Through :ref:`global patching <global_patching>` to enable patching for your scikit-learn installation for all further runs::

>     python sklearnex.glob patch_sklearn

Reviewer: Lets' also add wording that those different ways for enabling our optimizations are interchangeable and can be combined. 
I would also add details section below for advanced users on how this functionality works technically

Author: something like this is already in the documentation
https://intel.github.io/scikit-learn-intelex/what-is-patching.html#patching

I will add technical information
